### DISCUSSION TITLE
How to reset screen params ? / clear screen params

### DISCUSSION THREAD

**Discussion by Engazan** [react-navigation/react-navigation#10271] [Q&A] [answered]
How to reset screen params when clicked on button tab ?

**what i do:**
- Screen 1 => naviate to Screen 3 ( button tab )
- in Screen 3 i have 0 params ( **correct** )
- back to Screen 1  ( button tab )
- Screen 1 => press button ( not button tab )
 - Press Button to call
 ```ts
 navigation.navigate('screen3', {scrollTo: 'ordersSettings'})
```
- Screen 3 has "scrollTo" param ( **correct** ) 
- back to Screen 1 ( button tab )
- navigate to Screen 3 ( button tab ), but it has 1 params ( **incorect** ) => HOW to clear this ? 

i found "navigationKey", so if i change it all get reset but changing it inside of screen is terrible without any hook
https://reactnavigation.org/docs/group/#navigationkey

**Comment by raajnadar** [comment, marked as answer]
https://github.com/react-navigation/react-navigation/issues/6915#issuecomment-591533068

check this thread for more info https://github.com/react-navigation/react-navigation/issues/6915

**Reply by Engazan** [reply]
> [#6915 (comment)](https://github.com/react-navigation/react-navigation/issues/6915#issuecomment-591533068)
> 
> check this thread for more info #6915

its fine if u have direct Component inside bottomTab "**component**" prop but i have "**createNativeStackNavigator**" as parameter, so when i unmount it ( i lose ability to navigate to nested screeen ), and "createNativeStackNavigator" doesnt have "**unmountOnBlur**" prop 


____

tested it and it seems to not work, so i m stuck with hot fix having 2 same screens with diff name

tryed "unmountOnBlur" also ```listeners={({navigation}) => ({blur: () => navigation.setParams({screen: undefined})})}```, ...

**Reply by raajnadar** [reply]
Can you share a repro? Un mount is not compulsory in your case, that was the solution for that issue, the setParams is the solution for your case.

**Reply by Engazan** [reply]
> Can you share a repro? Un mount is not compulsory in your case, that was the solution for that issue, the setParams is the solution for your case.

scroll param is fixed with

```
listeners={({navigation}) => ({blur: () => navigation.setParams({scrollTo: undefined})})}
```

then combination with "useFocusEffect" does a trick to track changes

thanks

**Reply by rohan-1898** [reply]
we can also use the below code to reset prams on tabPress always
```
listeners={({ navigation }) => ({
        tabPress: () => navigation.setParams({
          webRoute: ROUTES.BENEFITS.route,
        }),
      })}
```

**Reply by connorpmullins** [reply]
Running into this as well. For anyone else reading I just want to drop in that this solution should also work:
```
    listeners={({ navigation }) => ({
      tabPress: e => {
            e.preventDefault();
            navigation.navigate(/*The Native Stack tied to the tab*/, {
              screen: /*the desired screen in the native stack*/,
              params: {/*desired params*/},
            });
```
The only benefit relative to setting the params is it will keep working without changes even if you add more params to a route / change param names, etc...
